### PR TITLE
macOS use constraints in utility bars

### LINKED ISSUE #3198: macOS - Filter bar: align along the symmetry axis
Align along the symmetry axis.

Screen Shot 2022-06-05 at 00 56 08

### PR DESCRIPTION
**Pull request by sweetppro**
Fixes #3198

### CODE REVIEW

**Comment by nevack** [comment]
Honestly, I have mixed feelings about that.

It's really great we can delegate all layout related calculations to AppKit.

What really bothers me it that managing/reviewing XIBs is really hard.

P.S. I hope we can adopt something like SwiftUI in the future.
I really like XML layouting in Android. It's easy and structured well, but XIBs is PITA

File: macosx/FilterBar.xib [hunk 1]
```diff
@@ -60,57 +51,100 @@
...
+                    <visibilityPriorities>
+                        <integer value="1000"/>
+                        <integer value="1000"/>
+                        <integer value="1000"/>
+                        <integer value="1000"/>
+                        <integer value="1000"/>
+                        <integer value="1000"/>
+                    </visibilityPriorities>
+                    <customSpacing>
+                        <real value="3.4028234663852886e+38"/>
```
[the remark this answers is not in the document]
Comment (sweetppro) [reply macosx/FilterBar.xib:133]: That is an odd value

**Comment by DevilDimon** [comment]
We can agree to only do layout in code for new views and rewrite old xibs to use code on significant refactors like this. Might be an issue on bigger views though

**Comment by sweetppro** [comment]
Does no one actually compile and run apps when reviewing?
A fresh build takes less than a min for me

File: macosx/FilterBar.xib (hunk 1 above)
[the remark this answers is not in the document]
Comment (sweetppro) [reply macosx/FilterBar.xib:133]: All values are generated by Xcode

**Comment by sweetppro** [comment]
> > Does no one actually compile and run apps when reviewing? A fresh build takes less than a min for me
> 
> Could you rephrase this? I'd like to understand your point, but am missing it: building doesn't seem to be necessary for either of the review comments, which are discussion on what layout strategy the project should use..?

Ah it was mentioned that it’s hard to review code that has been generated by Xcode. But this PR is a UI change, and I think it should be reviewed manually by compiling and testing rather than eyeballing the code changes.

**Comment by sweetppro** [comment]
The reason I submitted the PR was to have less code,  not just a refactor. Autolayout is a wonderful code free tool when done vie interface builder(Xcode), it would be a shame not to use it.

**Comment by ckerr** [comment]
Gotcha. Thanks for the clarification.

**Comment by ckerr** [comment]
@nevack @DevilDimon it sounds like you two would generally avoid more xib use in the mac client due to its complexity, is that correct?

I'm tryng to understand what the resolution should be for this PR.

**Comment by nevack** [comment]
> @nevack @DevilDimon it sounds like you two would generally avoid more xib use in the mac client due to its complexity, is that correct?
> 
> I'm tryng to understand what the resolution should be for this PR.

IMHO, XIBs are really hard to manage.
1. They are difficult to review.
2. It's impossible to merge in case of conflicts.
3. They are touched by Xcode every time I open them in some different env, like on external display.

I don't have experience in code-only UI.
I want to listen to some other opinions.

Regarding this PR - LGTM.
We cannot do anything about using XIBs right now.
